### PR TITLE
(components) feature: upgraded the Modal to @wethegit/react-modal v3 

### PR DESCRIPTION
**Pull request by pravton**
## Description

Upgraded the Modal component to use the latest version (v3) of @wethegit/react-modal which addresses the issue #318

Task: https://wethecollective.teamwork.com/app/tasks/22769488

### CODE REVIEW

**Review by andrewrubin — CHANGES_REQUESTED** [review]
Thanks for the updates @pravton! Couple notes from me, but we also need to clean up the docs:

This sentence is incorrect (there's no `trigger` prop on the Modal):
> _The <Modal> component takes a trigger prop that is a function which receives a toggle function as its only argument and should return the element that will trigger the modal._

Can we just re-word some of these docs to explain the usage better, i.e. how the `useModal` hook returns the props required for the Modal itself (`toggle` and `triggerRef`), as well as the `isOpen` state.

File: packages/wethegit-components/src/components/modal/modal.tsx
[the remark this answers is not in the document]
Comment (pravton) [reply packages/wethegit-components/src/components/modal/modal.tsx:1]: I agree and I like the idea of being explicit in accepting the renderTo prop. So I updated this with the renderRo Prop.

File: packages/wethegit-components/src/components/modal/modal.stories.tsx
[the remark this answers is not in the document]
Comment (pravton) [reply packages/wethegit-components/src/components/modal/modal.stories.tsx:1]: Thanks, great catch! I have also removed this!

**Comment by pravton** [comment]
@andrewrubin I have also updated the readme and the example and ready for you when you have a moment!

**Comment by andrewrubin** [comment]
It's looking good to me @pravton. I think we're ready for a changeset on this one 🎉

**Review by andrewrubin — APPROVED** [review]

**Pull request merged**